common/web/gesture-recognizer/src/engine/headless/gestureSource.ts

```diff
@@ -167,25 +203,61 @@ export class GestureSourceSubview<HoveredItemType> extends GestureSource<Hovered
    * @param initialHoveredItem  The initiating event's original target element
    * @param isFromTouch    `true` if sourced from a `TouchEvent`; `false` otherwise.
    */
-  constructor(source: GestureSource<HoveredItemType>, startAtEnd: boolean, preserveBaseItem: boolean) {
-    super(source.rawIdentifier, source.isFromTouch);
+  constructor(
+    source: GestureSource<HoveredItemType>,
+    configStack: typeof GestureSource.prototype['recognizerConfigStack'],
+    startAtEnd: boolean,
+    preserveBaseItem: boolean
+  ) {
+    let mayUpdate = true;
+    let start = 0;
+    let length = source.path.coords.length;
+    if(source instanceof GestureSourceSubview) {
+      const expectedLength = source._baseStartIndex + source.path.coords.length;
+      start = source._baseStartIndex;
+      // Check against the full remaining length of the original source; does
+      // the subview provided to us include its source's most recent point?
+      const sampleCountSinceStart = source.baseSource.path.coords.length;
+      if(expectedLength != start + sampleCountSinceStart) {
+        mayUpdate = false;
+      }
+    }
+
+    super(source.rawIdentifier, configStack, source.isFromTouch);
 
     const baseSource = this._baseSource = source instanceof GestureSourceSubview ? source._baseSource : source;
 
+    /**
+     * Provides a coordinate-system translation for source subviews.
+     * The base version still needs to use the original coord system, though.
+     */
+    const translateSample = (sample: InputSample<HoveredItemType>) => {
+      const translation = this.recognizerTranslation;
+      // Provide a coordinate-system translation for source subviews.
+      // The base version still needs to use the original coord system, though.
+      return {...sample, targetX: sample.targetX - translation.x, targetY: sample.targetY - translation.y};
+    }
+
     // Note: we don't particularly need subviews to track the actual coords aside from
     // tracking related stats data.  But... we don't have an "off-switch" for that yet.
     let subpath: GesturePath<HoveredItemType>;
 
     // Will hold the last sample _even if_ we don't save every coord that comes through.
     const lastSample = source.path.stats.lastSample;
 
+    // Are we 'chop'ping off the existing path or preserving it?  This sets the sample-copying
+    // configuration accordingly.
     if(startAtEnd) {
-      subpath = new GesturePath<HoveredItemType>();
-      if(lastSample) {
-        subpath.extend(lastSample);
-      }
+      this._baseStartIndex = start = Math.max(start + length - 1, 0);
+      length = length > 0 ? 1 : 0;
     } else {
-      subpath = source.path.clone();
+      this._baseStartIndex = start;
+    }
+
+    subpath = new GesturePath<HoveredItemType>();
+    for(let i=0; i < length; i++) {
+      const index = start + i;
+      subpath.extend(translateSample(baseSource.path.coords[index]));
```

Comment:
```suggestion
      subpath.extend(translateSample(baseSource.path.coords[start + i]));
```

Not sure the additional variable gives us much here :grin: